connect-inject/container_sidecar.go

```diff
@@ -39,11 +64,19 @@ func (h *Handler) containerSidecar(pod *corev1.Pod) corev1.Container {
 			"envoy",
 			"--config-path", "/consul/connect-inject/envoy-bootstrap.yaml",
 		},
-	}
+	}, nil
 }
 
-const sidecarPreStopCommand = `
+const sidecarPreStopCommandTpl = `
 export CONSUL_HTTP_ADDR="${HOST_IP}:8500"
 /consul/connect-inject/consul services deregister \
+  {{- if . }}
+  -token-file="/consul/connect-inject/acl-token" \
+  {{- end }}
   /consul/connect-inject/service.hcl
+{{- if . }}
+&& /consul/connect-inject/consul \
```

Comment:
```suggestion
&& /consul/connect-inject/consul logout \
```

Reply:
With the other commands, it would not correctly read the `-token-file` flag coming after the main part of the command. Is that not the case for this one?

Reply:
flags always have to come after the subcommand from what i can tell

Reply:
Getting things confused. I need a nap.